### PR TITLE
job-list: minor code consistency cleanup

### PR DESCRIPTION
**Pull request by chu11**
technically is `job-list` and things related to `job-list`

### CODE REVIEW

File: src/common/libjob/jj.c [hunk 1]
```diff
@@ -16,6 +16,8 @@
 #include <string.h>
 #include <jansson.h>
 
+#include "src/common/libccan/ccan/str/str.h"
```
Comment (grondo) [inline src/common/libjob/jj.c:19]: `-I$(top_srcdir)/src/common/libccan` is specified in the Makefile, so only `ccan/str/str.h` is required here.

File: src/common/libjob/test/sign_none.c [hunk 2]
```diff
@@ -20,6 +20,7 @@
 #include "src/common/libtap/tap.h"
 #include "src/common/libjob/sign_none.h"
 #include "src/common/libccan/ccan/base64/base64.h"
+#include "src/common/libccan/ccan/str/str.h"
```
Comment (grondo) [inline src/common/libjob/test/sign_none.c:23]:
```suggestion
#include "ccan/str/str.h"
```

File: src/modules/job-list/job_state.c [hunk 3]
```diff
@@ -25,6 +25,7 @@
 #include "src/common/libutil/grudgeset.h"
 #include "src/common/libjob/job_hash.h"
 #include "src/common/libidset/idset.h"
+#include "src/common/libccan/ccan/str/str.h"
```
Comment (grondo) [inline src/modules/job-list/job_state.c:28]:
```suggestion
#include "ccan/str/str.h"
```

File: src/modules/job-list/job_util.c [hunk 4]
```diff
@@ -19,6 +19,7 @@
 #include <assert.h>
 
 #include "src/common/libutil/errno_safe.h"
+#include "src/common/libccan/ccan/str/str.h"
```
Comment (grondo) [inline src/modules/job-list/job_util.c:22]:
```suggestion
#include "ccan/str/str.h"
```

**Review by grondo — APPROVED** [review]
Looks like good cleanup to me. Maybe change the PR title to be more release-notes worthy, and I noted that the full path path to str.h doesn't need to be specified where `-I/path/to/libccan` is already specified in `Makefile.am`s

**Comment by chu11** [comment]
@grondo thanks for the review.  updated the title and tweaked per your recommendations.  There were a few other "src/common/libccan/cccan/..." includes in some files in `libjjob`, so I just tweaked those includes as well.

**Pull request merged**